ocs_sample_library_preview/Security/UserInvitation.py

```diff
@@ -0,0 +1,177 @@
+        self.Id = id
+        self.ExpiresDateTime = expires_date_time
+        self.Issued = issued
+        self.Expires = expires
+        self.Accepted = accepted
+        self.State = state
+        self.SendInvitation = send_invitation
+        self.IdentityProviderId = identity_provider_id
+        self.TenantId = tenant_id
+        self.UserId = user_id
+    
+    @property
+    def Id(self) -> str:
+        return self.__id
+
+    @Id.setter
+    def Id(self, value: str):
+        self.__id = value
+
+    @property
+    def ExpiresDataTime(self) -> str:
+        return self.__expires_date_time
+
+    @ExpiresDataTime.setter
+    def ExpiresDataTime(self, value: str):
+        self.__expires_date_time = value
+
+    @property
+    def Issued(self) -> str:
+        return self.__issued
+
+    @Issued.setter
+    def Issued(self, value: str):
+        self.__issued = value
+    
+    @property
+    def Expires(self) -> str:
+        return self.__expires
+
+    @Expires.setter
+    def Expires(self, value: str):
+        self.__expires = value
+
+    @property
+    def Accepted(self) -> str:
+        return self.__accepted
+
+    @Accepted.setter
+    def Accepted(self, value: str):
+        self.__accepted = value
+
+    @property
+    def State(self) -> UserInvitationState:
+        return self.__state
+
+    @State.setter
+    def State(self, value: UserInvitationState):
+        self.__state = value
+
+    @property
+    def SendInvitation(self) -> str:
+        return self.__send_invitation
+
+    @SendInvitation.setter
+    def SendInvitation(self, value: str):
+        self.__send_invitation = value
+
+    @property
+    def IdentityProviderId(self) -> str:
+        return self.__identity_provider_id
+
+    @IdentityProviderId.setter
+    def IdentityProviderId(self, value: str):
+        self.__identity_provider_id = value
+
+    @property
+    def TenantId(self) -> str:
+        return self.__tenant_id
+
+    @TenantId.setter
+    def TenantId(self, value: str):
+        self.__tenant_id = value
+
+    @property
+    def UserId(self) -> str:
+        return self.__user_id
+
+    @UserId.setter
+    def UserId(self, value: str):
+        self.__user_id = value
+
+    def toJson(self):
+      return json.dumps(self.toDictionary())
+
+    def toDictionary(self):
+        result = {}
+
+        if self.Id is not None:
+            result['Id'] = self.Id
+
+        if self.ExpiresDateTime is not None:
+            result['ExpiresDateTime'] = self.ExpiresDateTime
+
+        if self.Issued is not None:
+            result['Issued'] = self.Issued
+
+        if self.Expires is not None:
+            result['Expires'] = self.Expires
+
+        if self.Accepted is not None:
+            result['Accepted'] = self.Accepted
+        
+        if self.State is not None:
+            result['State'] = self.State
```

Comment:
do we need to say either self.State.Name or self.State.Value?

Reply:
I think that you are right that it should be ```self.State.value``` so I have changed it.